aibolit/ast_framework/computed_fields_registry.py

```diff
@@ -0,0 +1,69 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 Aibolit
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from collections import defaultdict
+import sys
+from typing import Dict, Callable, Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from aibolit.ast_framework import ASTNode, ASTNodeType  # noqa: F401
+
+
+class _ComputedFieldsRegistry:
```

Comment:
Do we have tests for that class?

Reply:
No, this class is quite simple. It register functions in dictionary, provide read access to its item and may clear it. Nothing much. If you consider to test the function which checks if we are in interactive mode or not, that it is auxiliary functionality, it just simplifies prototyping. If it get broken and will affect production use case, we will catch it with `test_ast_node.py`. After all to test such functionality `unittest` package is not enough. We have to execute our code in different environment.